### PR TITLE
Feature mes 3625 schema for category c

### PR DESCRIPTION
**Pull request by rossfauldsbjss**
Introduction of a new schema for category C vehicles which includes:

addingbusinessDetails
adding code78Present (automatic indicator on license)
adding reverseLeft as manoeuvre and removing all other non-applicable manoeuvres
removing instructorReg
removing eyesight test
removing school car/ dual controls
refactor of vehicleChecks for multiple showMe/tellMe questions

### CODE REVIEW

File: mes-test-schema/categories/C/index.json [hunk 1]
```diff
@@ -0,0 +1,1749 @@
...
+        },
+        "angledStartControlledStop": {
+          "description": "Indicates whether or not this test requirement was carried out",
+          "type": "boolean"
+        },
+        "upHillStart": {
+          "description": "Indicates whether or not this test requirement was carried out",
+          "type": "boolean"
+        },
+        "downnHillStart": {
```
Comment (MarkPatt89) [inline mes-test-schema/categories/C/index.json]: typo

File: mes-test-schema/categories/C/index.json [hunk 2]
```diff
@@ -0,0 +1,1749 @@
...
+        "selected": {
+          "$ref": "#/definitions/manoeuvreIndicator"
+        },
+        "fault": {
+          "$ref": "#/definitions/manoeuvreOutcome"
+        },
+        "faultComments": {
+          "$ref": "#/definitions/faultComments"
+        }
+      },
```
Comment (MarkPatt89) [inline mes-test-schema/categories/C/index.json]: I don't think you need controlled stop according to this: https://wiki.dvsacloud.uk/display/MES/voc_5.0+Test+report

Comment (rossfauldsbjss) [reply mes-test-schema/categories/C/index.json]: removed

File: mes-test-schema/categories/C/index.json (hunk 1 above)
Comment (rossfauldsbjss) [reply mes-test-schema/categories/C/index.json]: fixed

File: mes-test-schema/categories/C/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,935 @@
...
+   */
+  normalStart2?: boolean;
+  /**
+   * Indicates whether or not this test requirement was carried out
+   */
+  upHill?: boolean;
+  /**
+   * Indicates whether or not this test requirement was carried out
+   */
+  downnHill?: boolean;
```
Comment (Cavalier07) [inline mes-test-schema/categories/C/index.d.ts]: typo in down

File: mes-test-schema/categories/C/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,935 @@
...
+  /**
+   * Category code for the test report
+   */
+  category: string;
+  journalData: JournalData;
+  activityCode: ActivityCode;
+  communicationPreferences?: CommunicationPreferences;
+  preTestDeclarations?: PreTestDeclarations;
+  accompaniment?: Accompaniment;
+  vehicleDetails?: VehicleDetails;
```
Comment (Cavalier07) [inline mes-test-schema/categories/C/index.d.ts]: I think you are missing vehicle height and length as it's used in the diagrams on the test report (Agree the same place a Mark to put it)

File: mes-test-schema/categories/C/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,935 @@
...
+  totalSeriousFaults?: number;
+  /**
+   * Count of the total number of dangerous faults incurred during the test
+   */
+  totalDangerousFaults?: number;
+}
+/**
+ * Finalisation of a successful test outcome
+ */
+export interface PassCompletion {
```
Comment (Cavalier07) [inline mes-test-schema/categories/C/index.d.ts]: can't see the code 78 field

File: mes-test-schema/categories/C/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,935 @@
...
+  ethnicityCode?: string;
+  /**
+   * Name of the business the candidate relates to
+   */
+  businessName?: string;
+  /**
+   * Telephone number of the business the candidate relates to
+   */
+  businessTelephone?: string;
+  businessAddress?: Address;
```
Comment (Cavalier07) [inline mes-test-schema/categories/C/index.d.ts]: These should have comments

File: mes-test-schema/categories/C/index.d.ts (hunk 3 above)
Comment (rossfauldsbjss) [reply mes-test-schema/categories/C/index.d.ts]: Fixed

File: mes-test-schema/categories/C/index.d.ts (hunk 4 above)
Comment (rossfauldsbjss) [reply mes-test-schema/categories/C/index.d.ts]: Had to speak to POs to confirm this was needed - seems that it maybe so have added width and height to vehicleDetails

File: mes-test-schema/categories/C/index.d.ts (hunk 5 above)
Comment (rossfauldsbjss) [reply mes-test-schema/categories/C/index.d.ts]: odd. code78Preseent is within passCompletion, maybe it didn't get pushed up. Can you check now.

File: mes-test-schema/categories/C/index.d.ts (hunk 6 above)
Comment (rossfauldsbjss) [reply mes-test-schema/categories/C/index.d.ts]: comments added

File: mes-test-schema/categories/C/index.d.ts (hunk 5 above)
Comment (Cavalier07) [reply mes-test-schema/categories/C/index.d.ts]: it's there now

**Pull request merged**